doc/source/dev_guide/aux_data.rst

```diff
@@ -0,0 +1,122 @@
+caching these files only when your component is used. This saves the user from
+wasting time and disk space downloading files they may never use.
+This functionality is made possible thanks to the
+`Pooch library <https://www.fatiando.org/pooch/latest/>`_.
+
+Downloaded files are stored in the directory configured by
+:ref:`data_dir_setting`.
+
+Adding download functionality
+-----------------------------
+
+The utility functions for data downloading include a two step process:
+
+1. **Registering**: Tell Satpy what files might need to be downloaded and used
+   later.
+2. **Retrieving**: Ask Satpy to download and store the files locally.
+
+Registering
+^^^^^^^^^^^
+
+Registering a file for downloading tells Satpy the remote URL for the file,
+and an optional hash. The hash is used to verify a successful download.
+Registering can also include a ``filename`` to tell Satpy what to name the
+file when it is downloaded. If not provided it will be determined from the URL.
+Once registered, Satpy can be told to retrieve the file (see below) by using a
+"cache key". Cache keys follow the general scheme of
+``<component_type>/<filename>`` (ex. ``readers/README.rst``).
+
+Satpy includes a low-level function and a high-level Mixin class for
+registering files. The higher level class is recommended for any Satpy
+component like readers, writers, and compositors. The lower-level
+:func:`~satpy.data_download.register_file` function can be used for any other
+use case.
+
+The :class:`~satpy.data_download.DataMixIn` class is automatically included
+in the :class:`~satpy.readers.yaml_reader.FileYAMLReader` and
+:class:`~satpy.writers.Writer` base classes. For any other component (like
+a compositor) you should include it as another parent class:
+
+.. code-block:: python
+
+    from satpy.data_download import DataDownloadMixin
+    from satpy.composites import GenericCompositor
+
+    class MyCompositor(GenericCompositor, DataDownloadMixin):
+        """Compositor that uses downloaded files."""
+
+        def __init__(self, name, url=None, known_hash=None, **kwargs):
+            super().__init__(name, **kwargs)
+            data_files = [{'url': url, 'known_hash': known_hash}]
+            self.register_data_files(data_files)
+
+However your code registers files, to be consistent it must do it during
+initialization so that the :func:`~satpy.data_download.find_registerable_files`.
+If your component isn't a reader, writer, or compositor then this function
+will need to be updated to find and load your registered files. See
+:ref:`offline_aux_downloads` below for more information.
+
+As mentioned, the mixin class is included in the base reader and writer class.
+To register files in these cases, include a ``data_files`` section in your
+YAML configuration file. For readers this would go under the ``reader``
+section and for writers the ``writer`` section. This parameter is a list
+of dictionaries including a ``url``, ``known_hash``, and optional
```

Comment:
it looks like `known_hash` is optional based on the following example? Also what is the advantage of having `known_hash: null` instead of just omitting it?

Reply:
PS: if see that my second question is answered in the docstring of the mixin class definition. However, for simplicity, maybe we can remove `known_hash: null` from the example here.

Reply:
Sounds good. Yeah the null was just show show where/how it would be there I guess. I'll remove it.